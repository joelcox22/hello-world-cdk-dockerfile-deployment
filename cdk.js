import { App } from 'aws-cdk-lib';
import { Stack, DockerfileDeployment } from '@pinecodes/cdk';

export const app = new App();

const stack = new Stack(app, 'hello-world');

new DockerfileDeployment(stack, 'hello-world', {
  containerPort: 80,
});
