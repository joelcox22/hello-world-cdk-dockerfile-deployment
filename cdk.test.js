import { app } from './cdk';
import { cdkAppSnapshotSerializer } from '@pinecodes/cdk';

expect.addSnapshotSerializer(cdkAppSnapshotSerializer);

test('snapshot', () => {
  expect(app).toMatchSnapshot();
});
