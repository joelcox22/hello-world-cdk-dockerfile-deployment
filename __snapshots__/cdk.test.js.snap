// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
--------- start snapshot of hello-world (account 849066399364) stack  --------
Tags: {}
Resources:
  helloworldtargetGroup072B8BB0:
    Type: AWS::ElasticLoadBalancingV2::TargetGroup
    Properties:
      HealthCheckEnabled: true
      HealthCheckPath: /
      HealthyThresholdCount: 2
      Matcher:
        HttpCode: '200'
      Port: 80
      Protocol: HTTP
      TargetGroupAttributes:
        - Key: deregistration_delay.timeout_seconds
          Value: '15'
        - Key: stickiness.enabled
          Value: 'false'
      TargetType: ip
      VpcId: vpc-12345
  helloworldtaskDefinitionTaskRoleA8F6AE15:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Action: sts:AssumeRole
            Effect: Allow
            Principal:
              Service: ecs-tasks.amazonaws.com
        Version: 2012-10-17
  helloworldtaskDefinition0C81A936:
    Type: AWS::ECS::TaskDefinition
    Properties:
      ContainerDefinitions:
        - Essential: true
          Image: !Sub >-
            849066399364.dkr.ecr.ap-southeast-2.\${AWS::URLSuffix}/cdk-pinecodes-container-assets-849066399364-ap-southeast-2:[HASH REMOVED FROM SNAPSHOT]
          LogConfiguration:
            LogDriver: awslogs
            Options:
              awslogs-group: !Ref helloworldtaskDefinitionappLogGroup32C744AC
              awslogs-stream-prefix: hello-world/hello-world
              awslogs-region: ap-southeast-2
          Name: app
          PortMappings:
            - ContainerPort: 80
              Protocol: tcp
      Cpu: '256'
      ExecutionRoleArn: !GetAtt helloworldtaskDefinitionExecutionRoleC93882F0.Arn
      Family: helloworldtaskDefinition84943F37
      Memory: '512'
      NetworkMode: awsvpc
      RequiresCompatibilities:
        - FARGATE
      TaskRoleArn: !GetAtt helloworldtaskDefinitionTaskRoleA8F6AE15.Arn
  helloworldtaskDefinitionappLogGroup32C744AC:
    Type: AWS::Logs::LogGroup
    UpdateReplacePolicy: Retain
    DeletionPolicy: Retain
  helloworldtaskDefinitionExecutionRoleC93882F0:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Action: sts:AssumeRole
            Effect: Allow
            Principal:
              Service: ecs-tasks.amazonaws.com
        Version: 2012-10-17
  helloworldtaskDefinitionExecutionRoleDefaultPolicyB2428EE2:
    Type: AWS::IAM::Policy
    Properties:
      PolicyDocument:
        Statement:
          - Action:
              - ecr:BatchCheckLayerAvailability
              - ecr:GetDownloadUrlForLayer
              - ecr:BatchGetImage
            Effect: Allow
            Resource: !Join 
              - ''
              - - 'arn:'
                - !Ref AWS::Partition
                - >-
                  :ecr:ap-southeast-2:849066399364:repository/cdk-pinecodes-container-assets-849066399364-ap-southeast-2
          - Action: ecr:GetAuthorizationToken
            Effect: Allow
            Resource: '*'
          - Action:
              - logs:CreateLogStream
              - logs:PutLogEvents
            Effect: Allow
            Resource: !GetAtt helloworldtaskDefinitionappLogGroup32C744AC.Arn
        Version: 2012-10-17
      PolicyName: helloworldtaskDefinitionExecutionRoleDefaultPolicyB2428EE2
      Roles:
        - !Ref helloworldtaskDefinitionExecutionRoleC93882F0
  helloworldcluster8A53E29D:
    Type: AWS::ECS::Cluster
  helloworldclusterDF6D02E6:
    Type: AWS::ECS::ClusterCapacityProviderAssociations
    Properties:
      CapacityProviders:
        - FARGATE
        - FARGATE_SPOT
      Cluster: !Ref helloworldcluster8A53E29D
      DefaultCapacityProviderStrategy: []
  helloworldserviceService90E54A69:
    Type: AWS::ECS::Service
    Properties:
      CapacityProviderStrategy:
        - CapacityProvider: FARGATE_SPOT
          Weight: 1
      Cluster: !Ref helloworldcluster8A53E29D
      DeploymentConfiguration:
        MaximumPercent: 200
        MinimumHealthyPercent: 50
      EnableECSManagedTags: false
      HealthCheckGracePeriodSeconds: 60
      LoadBalancers:
        - ContainerName: app
          ContainerPort: 80
          TargetGroupArn: !Ref helloworldtargetGroup072B8BB0
      NetworkConfiguration:
        AwsvpcConfiguration:
          AssignPublicIp: DISABLED
          SecurityGroups:
            - !GetAtt helloworldserviceSecurityGroupDAB172A9.GroupId
          Subnets:
            - p-12345
            - p-67890
      TaskDefinition: !Ref helloworldtaskDefinition0C81A936
    DependsOn:
      - helloworldlistenerRule27A0E55C
  helloworldserviceSecurityGroupDAB172A9:
    Type: AWS::EC2::SecurityGroup
    Properties:
      GroupDescription: hello-world/hello-world/service/SecurityGroup
      SecurityGroupEgress:
        - CidrIp: 0.0.0.0/0
          Description: Allow all outbound traffic by default
          IpProtocol: '-1'
      VpcId: vpc-12345
  helloworldserviceSecurityGroupfromhelloworldlistenerSecurityGroupsg123456789012FFF34F768055D8D323:
    Type: AWS::EC2::SecurityGroupIngress
    Properties:
      IpProtocol: tcp
      Description: Load balancer to target
      FromPort: 80
      GroupId: !GetAtt helloworldserviceSecurityGroupDAB172A9.GroupId
      SourceSecurityGroupId: sg-12345
      ToPort: 80
  helloworldpriorityhandlerServiceRole34911B70:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Action: sts:AssumeRole
            Effect: Allow
            Principal:
              Service: lambda.amazonaws.com
        Version: 2012-10-17
      ManagedPolicyArns:
        - !Join 
          - ''
          - - 'arn:'
            - !Ref AWS::Partition
            - ':iam::aws:policy/service-role/AWSLambdaBasicExecutionRole'
  helloworldpriorityhandlerServiceRoleDefaultPolicy4155F1CE:
    Type: AWS::IAM::Policy
    Properties:
      PolicyDocument:
        Statement:
          - Action: elasticloadbalancing:DescribeRules
            Effect: Allow
            Resource: '*'
        Version: 2012-10-17
      PolicyName: helloworldpriorityhandlerServiceRoleDefaultPolicy4155F1CE
      Roles:
        - !Ref helloworldpriorityhandlerServiceRole34911B70
  helloworldpriorityhandlerE3B799B4:
    Type: AWS::Lambda::Function
    Properties:
      Code:
        S3Bucket: cdk-pinecodes-assets-849066399364-ap-southeast-2
        S3Key: >-
          hello-world/[HASH REMOVED FROM SNAPSHOT].zip
      Role: !GetAtt helloworldpriorityhandlerServiceRole34911B70.Arn
      Architectures:
        - arm64
      Environment:
        Variables:
          AWS_NODEJS_CONNECTION_REUSE_ENABLED: '1'
      Handler: index.handler
      Runtime: nodejs16.x
    DependsOn:
      - helloworldpriorityhandlerServiceRoleDefaultPolicy4155F1CE
      - helloworldpriorityhandlerServiceRole34911B70
  helloworldpriorityhandlerLogRetention2A429690:
    Type: Custom::LogRetention
    Properties:
      ServiceToken: !GetAtt LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A.Arn
      LogGroupName: !Join 
        - ''
        - - /aws/lambda/
          - !Ref helloworldpriorityhandlerE3B799B4
      RetentionInDays: 7
  helloworldpriorityproviderframeworkonEventServiceRole3D1A3F62:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Action: sts:AssumeRole
            Effect: Allow
            Principal:
              Service: lambda.amazonaws.com
        Version: 2012-10-17
      ManagedPolicyArns:
        - !Join 
          - ''
          - - 'arn:'
            - !Ref AWS::Partition
            - ':iam::aws:policy/service-role/AWSLambdaBasicExecutionRole'
  helloworldpriorityproviderframeworkonEventServiceRoleDefaultPolicyBA4E3FDB:
    Type: AWS::IAM::Policy
    Properties:
      PolicyDocument:
        Statement:
          - Action: lambda:InvokeFunction
            Effect: Allow
            Resource:
              - !GetAtt helloworldpriorityhandlerE3B799B4.Arn
              - !Join 
                - ''
                - - !GetAtt helloworldpriorityhandlerE3B799B4.Arn
                  - ':*'
        Version: 2012-10-17
      PolicyName: >-
        helloworldpriorityproviderframeworkonEventServiceRoleDefaultPolicyBA4E3FDB
      Roles:
        - !Ref helloworldpriorityproviderframeworkonEventServiceRole3D1A3F62
  helloworldpriorityproviderframeworkonEvent518EFA67:
    Type: AWS::Lambda::Function
    Properties:
      Code:
        S3Bucket: cdk-pinecodes-assets-849066399364-ap-southeast-2
        S3Key: >-
          hello-world/[HASH REMOVED FROM SNAPSHOT].zip
      Role: !GetAtt helloworldpriorityproviderframeworkonEventServiceRole3D1A3F62.Arn
      Description: >-
        AWS CDK resource provider framework - onEvent
        (hello-world/hello-world/priority/provider)
      Environment:
        Variables:
          USER_ON_EVENT_FUNCTION_ARN: !GetAtt helloworldpriorityhandlerE3B799B4.Arn
      Handler: framework.onEvent
      Runtime: nodejs14.x
      Timeout: 900
    DependsOn:
      - >-
        helloworldpriorityproviderframeworkonEventServiceRoleDefaultPolicyBA4E3FDB
      - helloworldpriorityproviderframeworkonEventServiceRole3D1A3F62
  helloworldpriorityproviderframeworkonEventLogRetention347E4B94:
    Type: Custom::LogRetention
    Properties:
      ServiceToken: !GetAtt LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A.Arn
      LogGroupName: !Join 
        - ''
        - - /aws/lambda/
          - !Ref helloworldpriorityproviderframeworkonEvent518EFA67
      RetentionInDays: 7
  helloworldpriorityresource21A2AAFD2:
    Type: Custom::ListenerRulePriority
    Properties:
      ServiceToken: !GetAtt helloworldpriorityproviderframeworkonEvent518EFA67.Arn
      listenerArn: >-
        arn:aws:elasticloadbalancing:us-west-2:123456789012:listener/application/my-load-balancer/50dc6c495c0c9188/f2f7dc8efc522ab2
    UpdateReplacePolicy: Delete
    DeletionPolicy: Delete
  helloworldlistenerRule27A0E55C:
    Type: AWS::ElasticLoadBalancingV2::ListenerRule
    Properties:
      Actions:
        - TargetGroupArn: !Ref helloworldtargetGroup072B8BB0
          Type: forward
      Conditions:
        - Field: host-header
          HostHeaderConfig:
            Values:
              - hello-world.pine.codes
      ListenerArn: >-
        arn:aws:elasticloadbalancing:us-west-2:123456789012:listener/application/my-load-balancer/50dc6c495c0c9188/f2f7dc8efc522ab2
      Priority: !GetAtt helloworldpriorityresource21A2AAFD2.priority
  helloworlddnsCEC9186C:
    Type: AWS::Route53::RecordSet
    Properties:
      Name: hello-world.pine.codes.
      Type: CNAME
      HostedZoneId: DUMMY
      ResourceRecords:
        - my-load-balancer-1234567890.us-west-2.elb.amazonaws.com
      TTL: '300'
  LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Action: sts:AssumeRole
            Effect: Allow
            Principal:
              Service: lambda.amazonaws.com
        Version: 2012-10-17
      ManagedPolicyArns:
        - !Join 
          - ''
          - - 'arn:'
            - !Ref AWS::Partition
            - ':iam::aws:policy/service-role/AWSLambdaBasicExecutionRole'
  LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB:
    Type: AWS::IAM::Policy
    Properties:
      PolicyDocument:
        Statement:
          - Action:
              - logs:PutRetentionPolicy
              - logs:DeleteRetentionPolicy
            Effect: Allow
            Resource: '*'
        Version: 2012-10-17
      PolicyName: >-
        LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB
      Roles:
        - !Ref LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB
  LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A:
    Type: AWS::Lambda::Function
    Properties:
      Handler: index.handler
      Runtime: nodejs14.x
      Code:
        S3Bucket: cdk-pinecodes-assets-849066399364-ap-southeast-2
        S3Key: >-
          hello-world/[HASH REMOVED FROM SNAPSHOT].zip
      Role: !GetAtt LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB.Arn
    DependsOn:
      - >-
        LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB
      - LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB
Parameters:
  BootstrapVersion:
    Type: AWS::SSM::Parameter::Value<String>
    Default: /cdk-bootstrap/pinecodes/version
    Description: >-
      Version of the CDK Bootstrap resources in this environment, automatically
      retrieved from SSM Parameter Store. [cdk:skip]
Rules:
  CheckBootstrapVersion:
    Assertions:
      - Assert: !Not 
          - Fn::Contains:
              - - '1'
                - '2'
                - '3'
                - '4'
                - '5'
              - !Ref BootstrapVersion
        AssertDescription: >-
          CDK bootstrap stack version 6 required. Please run 'cdk bootstrap'
          with a recent version of the CDK CLI.

--------- end of stack snapshot hello-world (account 849066399364) ---------
`;
